ts/Series/MapPoint/MapPointSeries.ts

```diff
@@ -159,6 +161,11 @@ class MapPointSeries extends ScatterSeries {
         }
         this.generatePoints();
 
+        if (this.bounds && this.getProjectedBounds) {
```

Comment:
Just a minor optimization - in the current state, the bounds will always be re-calculated when translate runs again. For example in this case, it is not necessary: https://jsfiddle.net/highcharts/Lw6vxy8j/

I think what we want to do is to run this only when the data has changed, so we can simply add a check for `isDirtyData`:

`if (this.bounds && this.getProjectedBounds && this.isDirtyData) {`

Reply:
[2nd bug from Rafał](http://jsfiddle.net/BlackLabel/j61z9gLe/) (start with empty data) suggests that checking `this.bounds` is not needed - when debugging, I changed the undefined to true to enter the if and the chart works fine. In short, if data is dirty and getProjectedBounds is available then it should run after clearing bounds (clearing undefined on an object is not raising any errors).
`if (this.getProjectedBounds && this.isDirtyData) {`
Tested locally and works fine. Doesn't resolve the problem with [processedData](https://github.com/highcharts/highcharts/issues/17057).

Reply:
Thank you Torstein and Kacper for the suggestions!